### DISCUSSION TITLE
using `ByteArrayCodec.INSTANCE` in `RedissonBasedProxyManager`

### DISCUSSION THREAD

**Discussion by Jaysok** [vladimir-bukhtoyarov/bucket4j#460] [Q&A] [answered]
Hi!

Upon testing the behavior of Bucket4j with Redisson, I discovered that `RedissonBasedProxyManager` uses `ByteArrayCodec.INSTANCE` instead of the codec configured in `RedissonClient`, which in my case is `JsonJacksonCodec.INSTANCE`. I expected to see the entries I configured in Redis in JSON format. However, they are encoded in binary which is not human readable. This seems to be due to the following code segment:

```java
// io.github.bucket4j.redis.redisson.cas.RedissonBasedProxyManager.java;

RFuture<Boolean> redissonFuture = commandExecutor.writeAsync(stringKey, ByteArrayCodec.INSTANCE, SET, stringKey, encodeByteArray(newData), "NX");
```

If my understanding is correct, the codec used here should match the codec configured in `RedissonClient`. Is this a deliberate design decision, or is it something that has not yet been developed?

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @Jaysok 

> I expected to see the entries I configured in Redis in JSON format. However, they are encoded in binary which is not human readable.

Wrapping-out byte array by quotas will not make it more readable, it will stay be array of bytes.
![image](https://github.com/bucket4j/bucket4j/assets/6761830/1963eb32-8360-48b1-935a-3745142687ac)

> If my understanding is correct, the codec used here should match the codec configured in RedissonClient.

No. Redisson API explicitly allows you to choose codec that is more performant/suitable for each particular case.

If you want to read stored data, you shoud to learn serialization/deserialization routine https://github.com/bucket4j/bucket4j/blob/8.9/bucket4j-core/src/main/java/io/github/bucket4j/distributed/remote/RemoteBucketState.java#L51
